### PR TITLE
feat: add possibility to use user-created routes

### PR DESCRIPTION
**Pull request by mmelko**
This PR adds `externalRoutes` to the CRD. Operator will iterate on the routes list and create annotation for the oauth servcie account so routes can be used.

### CODE REVIEW

File: pkg/resources/service_account_test.go
Comment (tadayosi) [inline pkg/resources/service_account_test.go:1]: No assertions for `serviceaccounts.openshift.io/oauth-redirectreference.*`?

File: deploy/crd/hawtio_v1alpha1_hawtio_crd.yaml
Comment (tadayosi) [inline deploy/crd/hawtio_v1alpha1_hawtio_crd.yaml:1]: Please make sure to update `bundle/manifests/hawt.io_hawtios.yaml` as well.

**Review by tadayosi — APPROVED** [review]
LGTM. Thanks!

File: pkg/apis/hawtio/v1alpha1/hawtio_types.go [hunk 1]
```diff
@@ -96,6 +96,8 @@ type HawtioRoute struct {
 	CertSecret corev1.LocalObjectReference `json:"certSecret,omitempty"`
 	// Ca certificate secret key selector
 	CaCert corev1.SecretKeySelector `json:"caCert,omitempty"`
+	// List of external route names that will be annotated by the operator to access the console using the routes
+	ExternalRoutes []string `json:"externalRoutes,omitempty"`
```
Comment (astefanutti) [inline pkg/apis/hawtio/v1alpha1/hawtio_types.go]: This is not strictly related to the default Route that's created by the operator. It effectively relates to the ServiceAccount as OAuth client, rather than the default Route. So I wonder whether it'd better be declared outside the `HawtioRoute` struct?

**Review by astefanutti — APPROVED** [review]
I'm still wondering what's the appropriate place for the new `externalRoutes` field, but I trust your judgement :) Other than that LGTM 👍🏼.

File: deploy/crd/hawtio_v1alpha1_hawtio_crd.yaml
Comment (mmelko) [reply deploy/crd/hawtio_v1alpha1_hawtio_crd.yaml:1]: updated. thanks!

File: pkg/apis/hawtio/v1alpha1/hawtio_types.go (hunk 1 above)
Comment (mmelko) [reply pkg/apis/hawtio/v1alpha1/hawtio_types.go]: thanks! You're right, I moved it outside.

**Review by astefanutti — APPROVED** [review]
